sweep/tx_input_set.go

```diff
@@ -30,6 +30,16 @@ const (
 	constraintsForce
 )
 
+// InputSet defines an interface that's responsible for filtering a set of
+// inputs that can be swept economically.
+type InputSet interface {
+	// Inputs returns the set of inputs that should be used to create a tx.
+	Inputs() []input.Input
```

Comment:
We do have `fn.Set[T]` if needed, but I think you can run into weird interactions as it'll then try to key according to just the interface value/pointer: https://go.dev/play/p/717be9e_6dG

So I think maybe that needs to be re-worked to take in a diff type/interface that can give it a more stable                    identifier.

Reply:
I literally cannot comprehend the playground you sent above. Doesn't this completely break map semantics? Is it that the `xs[n]` results in the *same* vtable pointer??

Reply:
ok this just blew my mind😂 I think it has to do with the empty structs, as they are sharing the same pointers. I added a field in the struct and now it behaves as expected: https://go.dev/play/p/zT5cASv8F0L
if we remove `bar` then `x1` and `x2` are the same thing...

Reply:
I swear to god that's a compiler bug as it breaks a key assumption that every constructor invocation is a distinct memory address. I think in the case of empty structs it's making the assumption that because it can't have distinct values, then all empty structs can share the same memory address since it can't really be mutated, but that's sketchy as hell.